### DISCUSSION TITLE
Why is Junit5 slow than Junit4?

### DISCUSSION THREAD

**Discussion by PichlerMathias** [mockito/mockito#3489] [Q&A]
I'm migrating our tests from JUnit 4 to JUnit 5, and I'm noticing a substantial increase in execution time for tests using Mockito. Specifically, empty tests take almost 100 times longer with JUnit 5.

In JUnit 4, an empty test using MockitoJUnitRunner completes in around 6ms:

```
@RunWith(MockitoJUnitRunner.class)
public class X {

    @Test
    public void test() {
    }
}
```

However, in JUnit 5, an equivalent empty test using MockitoExtension takes approximately 600ms:

```
@ExtendWith(MockitoExtension.class)
public class X {

    
    @Test
    public void test() {
    }

}
```

Is there any way of making the Extension faster?

**Comment by srcimon** [comment]
This doesn't scale with number of tests. Initial scan for tests may be slower. This may also be dependent on the ide you are using.
